### PR TITLE
chore: update example tests, test on ci

### PR DESCRIPTION
**Pull request by caarlos0**
- run tests on examples on CI too
- update teatest

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,6 +32,10 @@ jobs:
         run: go build -v ./...
         working-directory: ./examples
 
+      - name: Test examples
```
Comment (bashbunni) [inline .github/workflows/build.yml:35]: Should we at some point have our workflows here pointing to `meta`? Like we do in Charm (https://github.com/charmbracelet/charm/blob/main/.github/workflows/build.yml#L7)

**Review by bashbunni — APPROVED** [review]
Nice! Thanks for tagging me to review, it's always helpful :)

File: examples/simple/main_test.go
Comment (bashbunni) [inline examples/simple/main_test.go:1]: There's another PR (#759 ) where this test isn't passing, but it passes here just fine

File: .github/workflows/build.yml (hunk 1 above)
Comment (caarlos0) [reply .github/workflows/build.yml:35]: not sure we can: the meta workflow just clone, build and test `./...`, we also do the testing in examples and tutorials... maybe if we change it to allow to pass the working directory 🤔 

anyway, will have a look on it in another PR 👍🏻

File: examples/simple/main_test.go
Comment (caarlos0) [reply examples/simple/main_test.go:1]: 🤔

**Pull request merged**